### ISSUE TITLE
Better documentation. 

### ISSUE THREAD

**Issue by iamdavidmt** [eco-infra/ecoinfra#15]
Look into better ways to document the tool and platform as a whole.

**Comment by rickwysocki** [comment]
Hi there! I'm looking to build out my portfolio of documentation writing and this seems like an interesting project. Can you give any details on the type of documentation you're looking for? E.g., an OpenAPI spec file, reference documentation, a User Guide, etc?

**Comment by rickwysocki** [comment]
Okay, I can try to take a look at it. I just forked the repository to see if I could get it running to test the app/that class, but I'm having a hard time finding where to create an account so I can get an API key. Can you direct me to the right spot?

**Comment by iamdavidmt** [comment]
Currently in closed beta, but looking to open registration in a few weeks. Maybe check out the test: https://github.com/eco-infra/ecoinfra/blob/master/tests/run.test.ts

**Comment by rickwysocki** [comment]
Gotcha, thanks! I'll start working on this and share some documentation/thoughts soon.

**Comment by rickwysocki** [comment]
Hey there: I've spent just a little bit of time detailing my understanding of the methods for this class and just wanted to check in to see whether this (screenshot below) is the direction you were hoping for with the documentation. Two questions:

- Did you want more of a line-by-line description of the code in the class? As you'll see below, I took more of a user-guide approach for the arg import/methods, but am happy to revise.
- The tables include an empty "use case" column. I think this would be useful but am hesitant to come up with use cases since I can't try the application myself yet. Do you think this is a useful idea and, if so, do you have any thoughts about use cases for those methods?

Of course, feel free to point out any errors I've made or any other feedback you have. Thanks for letting me work on this!

<img width="932" alt="Screenshot 2024-03-31 at 10 49 17 AM" src="https://github.com/eco-infra/ecoinfra/assets/16529152/455ec89d-2f69-4767-a141-c295d02a51dd">

**Comment by iamdavidmt** [comment]
This looks great @rickwysocki - I suggest we create a `docs` directory and we can put documentation in there.

**Comment by rickwysocki** [comment]
Okay, I just submitted a PR. Depending on the amount of documentation you anticipate, I would recommend setting up a documentation site for organizing content. I did a quick setup of that on a different branch locally (using Jekyll) and included a screenshot of the result below. If you think that's something you'd be interested in, I'd be happy to get the docs site configured. I believe it should be able to be run out of the repository using a /docs folder as the source for GitHub pages.
![2024-04-01_12-01-05](https://github.com/eco-infra/ecoinfra/assets/16529152/fd887cb9-edcb-4655-9604-de01f3682031)
